### PR TITLE
Add plot_structure function using Makie.jl

### PR DESCRIPTION
**Pull request by timoleistner**
I've proposed my plotting function while leaving your code untouched. Either keep both and rename accordingly (plot_structure_makie() or something) or remove your code.
The circular layout looks weird but imo its a useless representation anyway.
TODO:
- Add CairoMakie to the Project.toml
- Update README.md Usage part; proposal:
    plot_structure("((...))"; sequence = "AAAAAAA")

### CODE REVIEW

File: src/plot_structure.jl [hunk 1]
```diff
@@ -122,3 +122,76 @@ plot_structure(pt::Pairtable;
...
+    )
+    Colorbar(
+        f[2, 1],
+        vertical = false,
+        colormap = colorscheme,
+        width = 500,
+        height = 12,
+    )
+    ax.aspect = DataAspect()
+    if isempty(targetdir)
```
[the remark this answers is not in the document]
Comment (timoleistner) [reply src/plot_structure.jl:190]: Perfect.

File: src/plot_structure.jl [hunk 2]
```diff
@@ -122,3 +122,76 @@ plot_structure(pt::Pairtable;
...
+    colorscheme::Symbol=:lightrainbow
+    )
+    fc = FoldCompound(sequence)
+    partfn(fc)
+    pt = Pairtable(structure)
+    x_coords, y_coords = plot_coords(structure; plot_type=layout_type)
+
+    markersize = 100 / sqrt(length(structure))
+    positions = [(a, b) for (a, b) in zip(x_coords, y_coords)]
+    pairs = map(x -> abs.(x), filter(
```
[the remark this answers is not in the document]
Comment (timoleistner) [reply src/plot_structure.jl:148]: Your suggestion is nicer. I didn't quite understand the pt object coming from the C library.

**Comment by marcom** [comment]
Many thanks!

**Comment by marcom** [comment]
Time to first plot is not ideal, it went from <1s (Luxor) to ~40s (Makie). I tried to improve it with precompile statements but no luck so far.

**Comment by timoleistner** [comment]
This seems to be a general julia problem..
I've tried with different backends of Makie (GLMakie & WGLMakie) without improvements.

**Comment by marcom** [comment]
Yes, i think there is a lot of work going on in Julia to fix this by doing more caching.  As far as i understand, at the moment, precompilation only saves time on inference and doesn't result in generated machine code being cached.

Some more reading if you are interested
https://discourse.julialang.org/t/precompile-why/78770/8
https://github.com/JuliaLang/julia/pull/44527
https://gist.github.com/JeffBezanson/dd86043ef867954bd7e2163ab66f8b4e

So sometime in the future it should all be better.

There might be something going on with invalidations (and Makie?), because adding precompile statements like they do in Makie actually made it even slower (50s vs 40s). Some more info here:
https://julialang.org/blog/2020/08/invalidations/

I probably won't have a lot of time to investigate this in the next few months.

Until then i'll open an issue in this repo so that we don't forget about it.

**Pull request merged**